cobrawap/pipeline/utils/snakefile.py

```diff
@@ -11,17 +11,18 @@
 SETTINGS_PATH = Path('~/.cobrawap/config').expanduser()
 
 def get_setting(key: str = None):
-    with open(SETTINGS_PATH, 'r') as f:
-        try:
+    try:
+        with open(SETTINGS_PATH, 'r') as f:
             settings = yaml.load(f)
             if not settings:
                 settings = {}
-        except Exception as e:
-            logger.warning(e)
-            raise FileNotFoundError("The settings file either doesn't exist "\
-                                    "or is not in proper JSON format! "\
-                                    "Try to run `cobrawap init` to set up the "\
-                                    "settings file.")
+    except Exception as e:
+        logger.warning(e)
+        raise FileNotFoundError("The settings file `~/.cobrawap/config` "\
+                                "either does not exist "\
+                                "or is not in proper JSON format! "\
+                                "Try to run `cobrawap init` to set up the "\
+                                "settings file.") from e
```

Comment:
What's the purpose of `from e`? 
The full exception is already displayed as warning